src/app/work-item/work-item.service.ts

```diff
@@ -163,10 +261,57 @@ export class WorkItemService {
     }
   }
 
+  /**
+   * Usage: Fetch an use by it's ID from the User-ID map
+   */
+  getUserById(userId: string): User {
+    if (userId in this.userIdMap) {
+      return this.userIdMap[userId];
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * This is to fetch locally fetched work items
+   * this is eventually be deprecated once work item linking is re-worked
+   */
+
   getLocallySavedWorkItems(): Promise<any> {
     return Promise.resolve(this.workItems);
   }
 
+  /**
+   * Usage: This method is to resolve the comments for a work item 
+   * This ethod is only called when a single item is fetched for the 
+   * details page.
+   * 
+   * @Input: WorkItem - wItem
+   */
+
+  resolveComments(wItem: WorkItem): void {
```

Comment:
It appears this function is only called for in memory operations. Is this correct?